lib/payment.rb

```diff
@@ -1,54 +1,26 @@
 # frozen_string_literal: true
 
-require 'net/http'
-require 'net/https'
 require 'salesforce/database'
+require 'stripe/gateway'
 require 'thank_you_mailer'
 
 class Payment
-  attr_accessor :request
-
-  def initialize(request, supporter_database = Salesforce::Database)
-    @request = request
+  def initialize(request_data, supporter_database = Salesforce::Database)
+    @request_data = request_data
     @supporter_database = supporter_database
   end
 
   def attempt
-    response = post('https://api.stripe.com/v1/charges')
-    if response.code == '200'
-      ThankYouMailer.send_email(request.email, request.name)
-      supporter_database.add_donation(request)
-      []
-    elsif response.code == '402'
-      [:card_error]
+    result = Stripe::Gateway.charge(request_data)
+    if result.okay?
```

Comment:
I'm not sure this logic is correct. It's possible for the Stripe payment to succeed, but the Salesforce update fails. In this scenario you probably still want to send the thank you email.